### PR TITLE
feat: add Windows Support

### PR DESCRIPTION
**Pull request by guitarrapc**
## tl;dr;

This PR add Windows support.

> I've confirmed forked setup-job-workspace-action run on Windows as same as macOS.

## Summary

Added 2 changes, one for action behavior change and other for actions-test workflow change to support Windows.

**Action change (symlink)**

  * WHAT: Current action missing support for Windows, due to symlink behavior on Windows. Because `fs.promises.symlink` default on Windows is NOT directory symlink and action create invalid symlink. 
  * HOW: Specifying fs.promises.symlink type `dir` which explicitly notify Windows create dir symlink. [As document said](https://nodejs.org/api/fs.html#fspromisessymlinktarget-path-type), symlinkOption is ignore on non-Windows environment, this effect nothing to other OS.
  * TEST: Added check file in Symlink test to probe behavior on Linux, macOS and Windows.

**actions-test workflow change (node.js current directory)**

  * WHAT: GitHub Workflow `actions-test` failed on `windows-latest` runner with `mv`, due to Node.js process's working directory locks FileIO on Windows. `mv` failed with process lock is Windows NTFS file system restriction. It means specifying `working-directry` paramter is required to Windows, but optional to Linux and macOS. 
  * HOW: Added `working-directory` input parameter to action.yml. This indicate nodejs to `process.chdir` and avoid file lock issue on Windows. `working-directory` automatically set `..` on Windows, and `''` for other by default.
  * TEST: Added `working-dirdtory` test to probe behavior on Linux, macOS and Windows.

## Should we set Default value of working-directory?

I think YES. Because of specifying `working-directory` belongs to internal affairs, we can set `..` for Windows, and `''` to macOS and Linux by default. This keep same api for all platform and free user from consider platform difference.

See detail for following behavior difference.

Linux. Nothing change to working-directory and no `cd`.
image

macOS. Nothing change to working-directory and no `cd`.
image

Windows. Automatically set `..` to working-directory, and `cd` exists.
image

## Result

**Action change**

before: Symlink "testRepo", which target test-testjob, is invalid on Windows. Also added test failed on Windows.

image

after: Symlink "testRepo", which target test-testjob, is valid on Windows. Added test success on Windows.

image

**actions-test workflow change**

before: `uses: ./` failed due to node.js process locks checkout repository and prevent from mv. Existing actions-test failed on `windows-latest`.

> https://github.com/guitarrapc/setup-job-workspace-action/actions/runs/5404846427/jobs/9819587074

image

after: `uses: ./` successfully complete and all test passes by specifying `with.working-directory: '..'`. Existing actions-test success on `windows-latest` without change.

image

### CODE REVIEW

File: .github/workflows/actions-test.yml [hunk 1]
```diff
@@ -89,7 +97,8 @@ jobs:
           prefix: "Prefix-"
           suffix: "-Suffix"
 
-      - name: Created workspace is not symlkink
+      - name: Created workspace is not symlkin
```
Comment (Kesin11) [inline .github/workflows/actions-test.yml:100]: This still seems to be typo. Can you please fix all `symlkin` in this file to `symlink`?
These were my mistake 😲

**Review by Kesin11 — COMMENTED** [review]
Awesome! Thanks for your contribution!
I don't have a Windows development environment, so I can't verify this locally, but tests run on Linux and macOS are successful, so I think no problem to merge it.

Can you please fix only typo?

File: .github/workflows/actions-test.yml (hunk 1 above)
Comment (guitarrapc) [inline .github/workflows/actions-test.yml:100]:
```suggestion
      - name: Created workspace is not symlkink
```

Comment (guitarrapc) [reply .github/workflows/actions-test.yml]: Thank you, fixed.

File: .github/workflows/actions-test.yml [hunk 2]
```diff
@@ -9,24 +9,27 @@ jobs:
   default_name:
     strategy:
       matrix:
-        runner: [ubuntu-latest, macos-latest]
+        runner: [ubuntu-latest, macos-latest, "windows-latest"]
```
Comment (srz-zumix) [inline .github/workflows/actions-test.yml:12]: Can you unify quoted and unquoted?

Comment (guitarrapc) [reply .github/workflows/actions-test.yml]: thank you, done.

**Review by srz-zumix — APPROVED** [review]

**Review by Kesin11 — APPROVED** [review]

**Comment by Kesin11** [comment]
@srz-zumix I can't merge it and blocked merge button with this message.
> You’re not [authorized](https://docs.github.com/articles/about-protected-branches/) to merge this pull request.

Can you merge it? I want to release next version after merge this pull-request.

**Comment by Kesin11** [comment]
@srz-zumix I just trigger release workflow. Can you review it?
https://github.com/DeNA/setup-job-workspace-action/actions/runs/5456739109

**Comment by Kesin11** [comment]
@guitarrapc We released [v2.1.0](https://github.com/DeNA/setup-job-workspace-action/releases/tag/v2.1.0) 🎉 
Thanks for your contribution!

**Comment by guitarrapc** [comment]
@Kesin11 Thank you!
